### DISCUSSION TITLE
Issues with Dreamcast adapter, no power to ESP32, with USB power was able to pair but now Dreamcast ports no longer working

### DISCUSSION THREAD

**Discussion by klunkee** [darthcloud/BlueRetro#189] [Q&A]
I made a PS2 BlueRetro adapter that is working fine and then I attempted to make a Dreamcast one. I found a video on YouTube where they were able to use an ESP32 with a single Dreamcast plug so I tried this first. When I plugged it in to the Dreamcast, there was no light on the ESP32 so I then tried connecting the unused pins for controllers 2-4 to the 3.3v pin. Was this a mistake to connect them all together? I then plugged this configuration in and again there was no light to the led. So then I just disconnected the 5v line and plugged it into external USB power and it did light up and I was able to pair a DS4 to it. However, there was no response on the Dreamcast to any inputs. The cable I used was a generic Dreamcast extension cable and I checked the internal wires for continuity with the controller port pins.

Unfortunately, I then noticed that the original Dreamcast controllers were not working when plugged into any port. I tried powercycling the Dreamcast several times and switching ports but none of them work now though the Dreamcast seems to be okay, I need to do more research but possibly a fuse was tripped inside and can be replaced.

I took a log of successfully pairing a DS4 to the ESP32 if that would be helpful. Does anyone have experience with making a Dreamcast adapter that can tell me where I possibly went wrong?

**Comment by darthcloud** [comment]
Get a polyfuse replacement like this:
https://www.digikey.ca/en/products/detail/littelfuse-inc/RXEF020-2/5015936

and follow this guide:
https://mmmonkey.co.uk/dreamcast-controller-port-fix-fuse-f1-replacement/

Before repluging your Blueretro make sure you got no short between 5v and gnd with multimeter continuity check. Also check for same between 3.3v and gnd.

Post pictures of your esp32 build were I can see connection easily.

**Reply by klunkee** [reply]
I will replace the fuse and see if that fixes the ports. It's a bit hard to get a good photo of the board. I used Dupont connectors to test it first. 

When I opened the cable up I checked the 5 wires and did not use the sense data wire from pin 4. I wired up the two data lines, 5v and gnd to the pins on the esp32 by the Dupont female connector and then soldered the other end to the Dreamcast cable wires. Shouldn't the esp32 receive power with just these 4 connections? I don't know why it would not even light up. 

I have the data pins for controllers 2-4 connected to Dupont terminals and then all connected together and then I connected that group to the 3.3v pin, is that right? 

I'll check again for shorts between gnd and power. It should be ok as nothing is directly soldered to the esp32.

**Reply by darthcloud** [reply]
yes wiring unused port pins to 3.3v is fine

**Reply by klunkee** [reply]
I managed to repair the controller board. I am embarrassed to say that I also had the pins reversed from misreading the diagram and I rewired it and it works great now.

Thank you for your help and this amazing project, it's really a dream come true for old consoles.

**Reply by darthcloud** [reply]
Enjoy!
